diff -uraN a/incsrch/incsrch.h b/incsrch/incsrch.h
--- a/incsrch/incsrch.h	2021-06-30 04:02:09.000000000 -0400
+++ b/incsrch/incsrch.h	2021-08-15 13:53:46.040132845 -0400
@@ -46,7 +46,7 @@
   #define _tstrlen(a) strlen((a))
   #define _tstrnlen(a, n) sstrnlen((a), (n))
   #if !defined(INT_PTR)
-    #define INT_PTR int
+    #define INT_PTR uintptr_t
   #endif
 #endif
 
@@ -122,7 +122,7 @@
     signed   char Y1;
     signed   char Selected;
     unsigned int  Flags;
-    INT_PTR       Data;  // Message number or pointer to string
+    uintptr_t       Data;  // Message number or pointer to string
   }DialogTemplateItem;
 
   extern int DialogFromTemplate(const TCHAR* sTitle,
diff -uraN a/incsrch/search.c b/incsrch/search.c
--- a/incsrch/search.c	2021-06-30 04:02:09.000000000 -0400
+++ b/incsrch/search.c	2021-08-15 14:16:10.912117413 -0400
@@ -90,17 +90,17 @@
 static void Help(void)
 {
     static const DialogTemplateItem DialogTemplate[11] = {
-        {DI_TEXT,  2,2, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("FAR Incremental Search plugin")               },
-        {DI_TEXT,  2,3, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("Version 2.1")                                 },
+        {DI_TEXT,  2,2, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("FAR Incremental Search plugin")               },
+        {DI_TEXT,  2,3, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("Version 2.1")                                 },
         {DI_TEXT,  2,4, FALSE,DIF_CENTERGROUP|DIF_BOXCOLOR|DIF_SEPARATOR,0                                                          },
-        {DI_TEXT,  2,5, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("copyright (c) 1999-2019, Stanislav V. Mekhanoshin")},
-        {DI_TEXT,  2,6, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("This program comes with ABSOLUTELY NO WARRANTY")},
-        {DI_TEXT,  2,7, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("This is free software, and you are welcome to redistribute it")},
-        {DI_TEXT,  2,8, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("under certain conditions.")},
-        {DI_TEXT,  2,9, FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("http://rampitec.us.to")                       },
+        {DI_TEXT,  2,5, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("copyright (c) 1999-2019, Stanislav V. Mekhanoshin")},
+        {DI_TEXT,  2,6, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("This program comes with ABSOLUTELY NO WARRANTY")},
+        {DI_TEXT,  2,7, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("This is free software, and you are welcome to redistribute it")},
+        {DI_TEXT,  2,8, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("under certain conditions.")},
+        {DI_TEXT,  2,9, FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("http://rampitec.us.to")                       },
         {DI_TEXT,  2,10,FALSE,DIF_CENTERGROUP|DIF_BOXCOLOR|DIF_SEPARATOR,0                                                          },
-        {DI_TEXT,  2,11,FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("Press F1 for help")                           },
-        {DI_BUTTON,2,12,FALSE,DIF_CENTERGROUP,                           (INT_PTR)_T("Ok")                                          }
+        {DI_TEXT,  2,11,FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("Press F1 for help")                           },
+        {DI_BUTTON,2,12,FALSE,DIF_CENTERGROUP,                           (uintptr_t)_T("Ok")                                          }
     };
     struct FarDialogItem DialogItems[12];
 
