--- box2d-3.1.1/CMakeLists.txt
+++ box2d-3.1.1/CMakeLists.txt
@@ -1,5 +1,4 @@
 ﻿cmake_minimum_required(VERSION 3.22)
-include(FetchContent)
 include(CMakeDependentOption)
 
 project(box2d
@@ -83,17 +82,6 @@
 			string(APPEND CMAKE_EXE_LINKER_FLAGS " ${EMSCRIPTEN_PTHREADS_LINKER_FLAGS}")
 		endif()
 
-		# Task system used in tests and samples
-		set(ENKITS_BUILD_EXAMPLES OFF CACHE BOOL "Build enkiTS examples")
-		FetchContent_Declare(
-			enkits
-			GIT_REPOSITORY https://github.com/dougbinks/enkiTS.git
-			GIT_TAG master
-			GIT_SHALLOW TRUE
-			GIT_PROGRESS TRUE
-		)
-		FetchContent_MakeAvailable(enkits)
-
 		add_subdirectory(shared)
 	endif()
 
