--- a/tools/gcc.jam
+++ b/tools/gcc.jam
@@ -223,7 +223,7 @@
         # right onces. Note that we do not provide a clean way to build a 32-bit
         # binary using a 64-bit compiler, but user can always pass -m32
         # manually.
-        local lib_path = $(root)/bin $(root)/lib $(root)/lib32 $(root)/lib64 ;
+        local lib_path = $(root)/bin $(root)/lib $(root)/lib32 $(root)/libx32 $(root)/lib64 ;
         if $(.debug-configuration)
         {
             ECHO "notice:" using gcc libraries "::" $(condition) "::" $(lib_path) ;
@@ -285,6 +285,1 @@
-        if [ MATCH "^(i.86|x86|powerpc|ppc|sparc)" : $(machine:L) ]
-        {
-            compile-link-flags $(condition)/<architecture>/<address-model>32 : -m32 ;
-            compile-link-flags $(condition)/<architecture>/<address-model>64 : -m64 ;
-        }
         if [ MATCH "^(s390)" : $(machine:L) ]
@@ -390,5 +391,2 @@
-    compile-link-flags <target-os>$(generic-os)/<architecture>s390x/<address-model>64 : -m64 ;
     local arch = power sparc x86 ;
-    compile-link-flags <target-os>$(generic-os)/<architecture>$(arch)/<address-model>32 : -m32 ;
-    compile-link-flags <target-os>$(generic-os)/<architecture>$(arch)/<address-model>64 : -m64 ;
 }